src/Badge.js

```diff
@@ -25,7 +25,7 @@ class Badge extends React.Component {
 
         return React.cloneElement(element, {
             className: 'mdl-badge',
-            'data-badge': this.props.text
+            'data-badge': this.props.text || undefined
```

Comment:
why not just `this.props.text`?
Does mdl render something if it's an empty string?